bitcoin/src/crypto/sighash.rs

```diff
@@ -1190,22 +1190,7 @@ impl<R: BorrowMut<Transaction>> SighashCache<R> {
     /// When the `SighashCache` is initialized with a mutable reference to a transaction instead of
     /// a regular reference, this method is available to allow modification to the witnesses.
     ///
-    /// This allows in-line signing such as
-    /// ```
-    /// use bitcoin::{absolute, transaction, Amount, Transaction, Script};
-    /// use bitcoin::sighash::{EcdsaSighashType, SighashCache};
-    ///
-    /// let mut tx_to_sign = Transaction { version: transaction::Version::TWO, lock_time: absolute::LockTime::ZERO, input: Vec::new(), output: Vec::new() };
-    /// let input_count = tx_to_sign.input.len();
-    ///
-    /// let mut sig_hasher = SighashCache::new(&mut tx_to_sign);
-    /// for inp in 0..input_count {
-    ///     let prevout_script = Script::new();
-    ///     let _sighash = sig_hasher.p2wpkh_signature_hash(inp, prevout_script, Amount::ONE_SAT, EcdsaSighashType::All);
-    ///     // ... sign the sighash
-    ///     sig_hasher.witness_mut(inp).unwrap().push(&Vec::new());
-    /// }
-    /// ```
+    /// For an example how to use this function please see `examples/sign-tx-*`.
```

Comment:
Why not both? Anyway, I'd like to improve the doc further.

> Allows modification of witnesses.
>
> As a lint against accidental changes to the transaction  that would invalidate the cache and signatures `SighashCache` borrows the `Transaction`, so that modifying it is not possible without hacks with `UnsafeCell` (which is hopefully a strong indication that something is wrong). However modifying witnesses never invalidates the cache and is actually useful - one usually wants to put the signature generated for an input into the witness of that input.
> This method allows doing exactly that if the transaction is owned by `SighashCache` or borrowed mutably.
>
> Simplified example:
>
> `<short example here>`
>
> For full signing example see `examples/sign-tx-*`.

It'd be great if we could make the examples into a link.

Side note: when writing this I realized using `Cell` or `RefCell` to bypass the protection is impossible, which is quite cool, I believe `GhostCell` is not usable either but not 100% sure about it.